### ISSUE TITLE
Error while getting metrics from Promentheus

### ISSUE THREAD

**Issue by DirkFries** [influxdata/influxdb#14261]
Hi @all,

I am experiencing some problems with InfluxDB while getting metrics from Prometheus.

__Steps to reproduce:__

Software involved:

InfluxDB v1.7.7 (git: 1.7 f8fdf652f348fc9980997fe1c972e2b79ddd13b0)
prometheus, version 2.10.0 (branch: HEAD, revision: d20e84d0fb64aff2f62a977adc8cfb656da4e286)

Prometheus is sending metrics to InfluxDB via a remote write configuration to http://127.0.0.1:8086/api/v1/prom/write?db=mydb

Every few seconds I see the following errors in the InfluxDB Output:

```sh
[httpd] 127.0.0.1 - - [04/Jul/2019:13:36:02 +0200] "POST /api/v1/prom/read?db=mydb HTTP/1.1" 200 0 "-" "Prometheus/2.10.0" e70aff1f-9e4f-11e9-845d-005056810c80 151 [panic:runtime error: invalid memory address or nil pointer dereference] goroutine 2237 [running]:
runtime/debug.Stack(0xc00a281ca0, 0xc000a84500, 0xbf3f9708abf2ca2e)
        /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
github.com/influxdata/influxdb/services/httpd.(*Handler).recovery.func1.1(0xc00a281ca0, 0xc000a84500, 0xbf3f9708abf2ca2e, 0x21fcb735fc0, 0x219c120, 0xc000122400, 0x15b6060, 0xc007bfe620)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1786 +0xcd
panic(0x110e9e0, 0x1ff0d00)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/influxdata/influxdb/services/httpd.(*Handler).servePromRead(0xc000122400, 0x7efcf329cf10, 0xc00a281ce0, 0xc000a84500, 0x0, 0x0)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1097 +0x30d
github.com/influxdata/influxdb/services/httpd.(*Handler).servePromRead-fm(0x7efcf329cf10, 0xc00a281ce0, 0xc000a84500, 0x0, 0x0)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:190 +0x5c
github.com/influxdata/influxdb/services/httpd.authenticate.func1(0x7efcf329cf10, 0xc00a281ce0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1578 +0x815
net/http.HandlerFunc.ServeHTTP(0xc0004d50e0, 0x7efcf329cf10, 0xc00a281ce0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.(*Handler).responseWriter.func1(0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1763 +0xab
net/http.HandlerFunc.ServeHTTP(0xc0004d5100, 0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.gzipFilter.func1(0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/gzip.go:23 +0x29d
net/http.HandlerFunc.ServeHTTP(0xc0004d5120, 0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.cors.func1(0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1705 +0xf3
net/http.HandlerFunc.ServeHTTP(0xc0004d5140, 0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.requestID.func1(0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1736 +0x179
net/http.HandlerFunc.ServeHTTP(0xc0004d5160, 0x15b12e0, 0xc00a281cc0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.(*Handler).logging.func1(0x15b12e0, 0xc00a281ca0, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1744 +0xdc
net/http.HandlerFunc.ServeHTTP(0xc0004d5180, 0x15b12e0, 0xc00a281ca0, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/services/httpd.(*Handler).recovery.func1(0x15b6060, 0xc007bfe620, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1800 +0x147
net/http.HandlerFunc.ServeHTTP(0xc0004d51a0, 0x15b6060, 0xc007bfe620, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:1964 +0x44
github.com/influxdata/influxdb/vendor/github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP(0xc0004d4480, 0x15b6060, 0xc007bfe620, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/vendor/github.com/bmizerany/pat/mux.go:117 +0x155
github.com/influxdata/influxdb/services/httpd.(*Handler).ServeHTTP(0xc000122400, 0x15b6060, 0xc007bfe620, 0xc000a84500)
        /go/src/github.com/influxdata/influxdb/services/httpd/handler.go:392 +0x247
net/http.serverHandler.ServeHTTP(0xc001c00410, 0x15b6060, 0xc007bfe620, 0xc000a84500)
        /usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc000176f00, 0x15b7ba0, 0xc0075ea0c0)
        /usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2851 +0x2f5
```
There is no striking output in Prometheus

__Expected behavior:__
No error messages

__Actual behavior:__
Error output, see above.

Does anyone have a hint for me what is going wrong ?

Thanks a lot !

**Comment by e-dard** [comment]
@DirkFries this will be fixed in 1.7.8, and is currently fixed in the 1.8 branch.

**Comment by sysword** [comment]
> @DirkFries this will be fixed in 1.7.8, and is currently fixed in the 1.8 branch.

Do we have a docker images for this?
